### DISCUSSION TITLE
Using Vercel build on GitHub Actions

### DISCUSSION THREAD

**Discussion by dgokcin** [vercel/vercel#8910] [Help]
Hello

I am having a problem with vercel's 15k file limit when deploying with the prebuild option. I would like to ask the difference between the following options:
- building on github actions and deploying with vercel deploy --prebuild(with a bot user)
- deploying from github actions with just vercel deploy(with a bot user)

Will I hit the 15k file limit if I use the second option and can I use the second option with just the 20USD pro plan without any additional charges?

Thanks

**Comment by leerob** [comment]
What are you building where you have 15k+ files, out of curiosity?

**Reply by dgokcin** [reply]
We are building a next.js app. We have no idea why the file count is so much. We tried debugging but could not find the reason for the increase.

**Reply by dgokcin** [reply]
I am just trying to find a way to integrate vercel with github actions without using the Git integration which charges you per committer.

**Reply by dgokcin** [reply]
In case someone comes across this problem, I worked around by using --archive=tgz option with vercel deploy. I will investigate the problem with the code that generates 15kfiles in .vercel/output directory. I will update here if I can figure out what the problem is.

**Reply by leerob** [reply]
Would love to see the root issue here, 15k files definitely does not sound expected in your app! 🙏
